src/main/java/com/jvm_bloggers/admin_panel/moderation/FetchingDataPanel.java

```diff
@@ -58,9 +44,8 @@ private AjaxLink createFetchBlogPostsDataButton(CustomFeedbackPanel feedback) {
             @Override
             public void onClick(AjaxRequestTarget target) {
                 log.debug("Fetching blog posts data clicked");
-                if (fetchingBlogPostExecutor.getActiveCount() == 0) {
-                    fetchingBlogPostExecutor
-                        .submit(blogPostsFetchingScheduler::checkRssForNewBlogPosts);
+                if (!blogPostsFetcher.isFetchingProcessInProgress()) {
```

Comment:
I agree, if `refreshPost` returns some request state then this method `isFetchingProcessInProgress` could me moved inside fetcher.